update tests, solve OutOfBounds issue, turn to es6

sandrabosk opened the pull request:
@mjarraya Please check the updated tests. 
Find here the proposed solution: https://gist.github.com/sandrabosk/1dc434be87b4efc03238e53ffae1376f

mjarraya commented on starter-code/test/test.js:
```diff
@@ -1,73 +1,72 @@
-  describe('#add(x)', function() {
-    var sl;
-    beforeEach(function(){
+  describe('#add(x)', () => {
```
This unit should also test the spec that the list of items is sorted

mjarraya commented on starter-code/test/test.js:
```diff
@@ -1,73 +1,72 @@
-  describe('#add(x) and get(i)', function() {
-    var sl;
-    beforeEach(function(){
+  describe('#add(x) and get(i)', () =>  {
```
these specs could be integrated in the unit that describes `add`

sandrabosk replied:
This test requires using `add()` and `get()` but in the moment we define `add()`, we still don't have `get()` defined. That's why I kept it after. Maybe I am not thinking in the right direction. Please let me know if I am wrong. Thank you.

mjarraya replied:
The students implement `add()` without the sorting, and the tests pass. Then they don't necessarily see what was wrong with their initial implementation (does it come from the `add()` ? from the `get()` ?) and have to backtrack to implement the `add` method correctly.

mjarraya commented on starter-code/test/test.js:
```diff
@@ -1,73 +1,72 @@
-        assert.equal(sl.get(i), foo*i);
-      }
+    it('should return an OutOfBounds exception if there is no element in that position', () => {
+      assert.throws(() => {
```
`assert.throws(() => { sl.get(4); }, Error, "OutOfBounds");`

It's not relevant to create a condition for equality of the return value of the method with the string `"undefined"` -which should never be reached-. `assert.throws()` only needs the function to be called with the right parameters, and we can pass in a third parameter to `.throws()` the error's name

mjarraya commented on starter-code/test/test.js:
```diff
@@ -77,97 +76,90 @@ describe('SortedList', function() {
-      sl = new SortedList();
+    it('should return an EmptySortedList exception if there is no elements in the array', () => {
+       assert.throws(() => {
+        if (sl.max([]) === 'undefined') {
```
`assert.throws(() => { sl.max([])) }, Error, "EmptySortedList");`

mjarraya commented on starter-code/test/test.js:
```diff
@@ -77,97 +76,90 @@ describe('SortedList', function() {
+
+    it('should return an EmptySortedList exception if there is no elements in the array', () => {
+      assert.throws(() => {
+        if (sl.min([]) === 'undefined') {
```
see above

mjarraya commented on starter-code/test/test.js:
```diff
@@ -77,97 +76,90 @@ describe('SortedList', function() {
-      assert.equal(sl.average(), 100);
+    it('should return an EmptySortedList exception if there are no elements', () => {
+      assert.throws(() => {
+        if (sl.avg([]) === 'undefined') {
```
see above

sandrabosk commented:
- updated tests that required a change
- removed global mocha installation (it's already using 6.x.x)

- **pending**: should additional test on _add ()_ be merged with _add()_?
